### DISCUSSION TITLE
Charset Auto Selector Return Type

### DISCUSSION THREAD

**Discussion by DahyeonWoo** [euphony-io/euphony#250] [Q&A] [answered]
I implemented the Charset Auto Selector.
However, I forgot to return to charset interface and couldn't implement it.
I have a question about it.

1. Is the return type Charset interface? Or should it be clear like ASCIICharset or UTF8Charset?
2. Would it be good to make CharsetBuilder class like PacketBuilder? Other implementation methods are also good, so I would appreciate it if you could give me some advice.

**Comment by yeonns** [comment, marked as answer]
@DahyeonWoo 
1. Is the return type Charset interface? Or should it be clear like ASCIICharset or UTF8Charset?
> Because `Charset` is super class, I think it is enough to return Charset interface. 
2. Would it be good to make CharsetBuilder class like PacketBuilder? Other implementation methods are also good, so I would appreciate it if you could give me some advice.
> Well, there can be many ways to implement, but I think one way is utilizing `map<Charset, HexVector>`. Then, we can sort by encoded result and also return Charset.

**Reply by designe** [reply]
@DahyeonWoo 
How about to use `typeid` comparison? https://www.oreilly.com/library/view/c-cookbook/0596007612/ch08s07.html
